File: server/router/gb28181/platform.go
package gb28181

import (
	"wvp-go/server/api/v1/gb28181"

	"github.com/gin-gonic/gin"
)

type PlatformRouter struct{}

var platformRouter = new(PlatformRouter)

func (p *PlatformRouter) InitPlatformRouter(Router *gin.RouterGroup) {
	platformRouterGroup := Router.Group("/platform")
	{
		platformRouterGroup.GET("/list", gb28181.GetPlatformList)
		platformRouterGroup.GET("/online", gb28181.GetOnlinePlatforms)
		platformRouterGroup.GET("/:id", gb28181.GetPlatform)
		platformRouterGroup.POST("", gb28181.CreatePlatform)
		platformRouterGroup.PUT("/:id", gb28181.UpdatePlatform)
		platformRouterGroup.DELETE("/:id", gb28181.DeletePlatform)
		platformRouterGroup.GET("/:id/channels", gb28181.GetPlatformChannels)
		platformRouterGroup.POST("/:id/sync", gb28181.SyncPlatformCatalog)
		platformRouterGroup.POST("/:id/start", gb28181.StartPlatformCascade)
		platformRouterGroup.POST("/:id/stop", gb28181.StopPlatformCascade)
	}
}

File: web/src/api/platform.js
import service from '@/utils/request'

export function getPlatformList(params) {
  return service({
    url: '/platform/list',
    method: 'get',
    params
  })
}

export function getOnlinePlatforms() {
  return service({
    url: '/platform/online',
    method: 'get'
  })
}

export function getPlatform(id) {
  return service({
    url: `/platform/${id}`,
    method: 'get'
  })
}

export function createPlatform(data) {
  return service({
    url: '/platform',
    method: 'post',
    data
  })
}

export function updatePlatform(id, data) {
  return service({
    url: `/platform/${id}`,
    method: 'put',
    data
  })
}

export function deletePlatform(id) {
  return service({
    url: `/platform/${id}`,
    method: 'delete'
  })
}

export function getPlatformChannels(id) {
  return service({
    url: `/platform/${id}/channels`,
    method: 'get'
  })
}

export function syncPlatformCatalog(id) {
  return service({
    url: `/platform/${id}/sync`,
    method: 'post'
  })
}

export function startPlatformCascade(id) {
  return service({
    url: `/platform/${id}/start`,
    method: 'post'
  })
}

export function stopPlatformCascade(id) {
  return service({
    url: `/platform/${id}/stop`,
    method: 'post'
  })
}
